gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py

```diff
@@ -0,0 +1,424 @@
+    def __init__(self, *args, **kwargs):
+        super(VPNSvcValidatorTestCase, self).__init__(*args, **kwargs)
+        self.dict_objects = test_vpn_data.MakeDictionaries()
+        self.plugin_rpc = vpn.VpnaasRpcSender(self.dict_objects.context,
+                                              self.dict_objects.sc)
+        self.valid_obj = vyos_vpn_driver.VPNSvcValidator(self.plugin_rpc)
+
+    def test_validate_active(self):
+        '''
+        Implements testcase for vpn driver's validate method to test in
+        success condition while making call to the service VM
+        '''
+
+        context = self.dict_objects._make_service_context()
+        svc = self.dict_objects._create_vpnservice_obj()['resource']
+        description = str(svc['description'])
+        description = description.split(';')
+        description[1] = 'tunnel_local_cidr=12.0.6.0/24'
+        description = ";".join(description)
+        svc.update({'description': description})
+
+        with mock.patch.object(self.plugin_rpc, "update_status") as mock_valid:
+            self.valid_obj.validate(context, svc)
+            mock_valid.assert_called_with(context,
+                                          self.dict_objects.vpn_vpnsvc_active)
+
+    def test_validate_error(self):
+        '''
+        Implements testcase for vpn driver's validate method to test in
+        fail condition while making call to the service VM
+        '''
+
+        context = self.dict_objects._make_service_context()
+        with mock.patch.object(self.plugin_rpc, "update_status") as mock_valid:
+            self.valid_obj.validate(
+                    context,
+                    self.dict_objects._create_vpnservice_obj()['resource'])
+            mock_valid.assert_called_with(
+                        context,
+                        self.dict_objects.vpn_vpnsvc_active)
+
+
+class RestApiTestCase(unittest.TestCase):
+    '''
+    Class which implements the testcases to test the vpn RestApi calls.
+    '''
+    def __init__(self, *args, **kwargs):
+        super(RestApiTestCase, self).__init__(*args, **kwargs)
+        self.rest_obj = vyos_vpn_driver.RestApi((
+                            test_vpn_data.MakeDictionaries().vm_mgmt_ip))
+        self.resp = mock.Mock()
+        self.resp = mock.Mock(status_code=200)
+        self.dict_objects = test_vpn_data.MakeDictionaries()
+        self.args = {'peer_address': '1.103.2.2'}
+        self.fake_resp_dict = {'status': None}
+        self.timeout = 30
+        self.data = {'data': 'data'}
+        self.j_data = json.dumps(self.data)
+
+    def test_post_success(self):
+        '''
+        Implements testcase for vpn drivers post method to test in
+        success condition while making call to the service VM
+        '''
+
+        self.resp = mock.Mock(status_code=200)
+        self.fake_resp_dict.update({'status': True})
+        with mock.patch.object(requests, 'post', return_value=self.resp) as (
```

Comment:
The REST API mock cases are tried in other tests right? What are we testing additionally by mocking the post method again?